osu.Game.Tournament/IPC/FileBasedIPC.cs

```diff
@@ -184,26 +181,25 @@ public bool SetIPCLocation(string path)
         /// <returns>Whether an IPC directory was successfully auto-detected.</returns>
         public bool AutoDetectIPCLocation() => SetIPCLocation(findStablePath());
 
-        private static bool ipcFileExistsInDirectory(string p) => p != null && File.Exists(Path.Combine(p, "ipc.txt"));
+        private static bool ipcFileExistsInDirectory([NotNullWhen(true)] string? p) => p != null && File.Exists(Path.Combine(p, "ipc.txt"));
```

Comment:
I don't think this is correct. Remove the weird usage of the `NotNullWhen` attribute.

Reply:
This is a [valid usage](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis#conditional-post-conditions-notnullwhen-maybenullwhen-and-notnullifnotnull) of the `NotNullWhen` attribute. It's useful for scenarios like this:

```cs
string? path;

if (ipcFileExistsInDirectory(path)
{
    // `path` is guaranteed to be non-null here.
}
````

Reply:
🤯 that's some very hidden logic. not sure i like it.

Reply:
Yeah that was from my suggestion earlier on. I wasn't sure initially about it myself but after I saw that the official docs said it was officially supported usage I decided to suggest it. Am fine with reverting though since I agree it's a bit weird when not in an `out` parameter usage.